net/service/handlers.go

```diff
@@ -840,3 +843,44 @@ func (service *OpenBazaarService) handleDisputeClose(p peer.ID, pmes *pb.Message
 
 	return nil, nil
 }
+
+func (service *OpenBazaarService) handleChat(p peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	log.Debugf("Received CHAT message from %s", p.Pretty())
+
+	// Unmarshall
+	chat := new(pb.Chat)
+	err := ptypes.UnmarshalAny(pmes.Payload, chat)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate
+	if len(chat.Subject) > 500 || len(chat.Message) > 20000 {
```

Comment:
This same logic is used elsewhere. It should be abstracted so if it changes we don't have to hunt all over. The error message is also ambiguous as to what property has the issue.